src/array_api_stubs/_draft/fft.py

```diff
@@ -60,7 +60,7 @@ def fft(
     Returns
     -------
     out: array
-        an array transformed along the axis (dimension) indicated by ``axis``. The returned array must have a complex floating-point data type determined by :ref:`type-promotion`.
+        an array transformed along the axis (dimension) indicated by ``axis``. If ``x`` has a complex floating-point data type, the returned array must have the same data type as ``x``. If ``x`` has a real-valued floating-point data type, the returned array must have a complex floating-point data type whose precision matches the precision of ``x`` (e.g., if ``x`` is ``float64``, then the returned array must have a ``complex128`` data type).
```

Comment:
I am against the change to accept real input here. I think in the past meetings we've decided to not allow that, and instead users must do explicit casting here.

Reply:
This is based on what was already specified (https://github.com/data-apis/array-api/pull/720/files#diff-31eaf0d090ecc4df2af89ab3d3a33b91ff8846925b579fd64bf065aea146a230R38).

Reply:
If it leaves room for alternative interpretation, we must shut down those loopholes, and here is not one 🙂

Reply:
I don't think this is an ambiguity. "floating-point data type" in the spec right now means "real or complex" (see https://data-apis.org/array-api/draft/API_specification/data_types.html#floating-point-data-types). Right now the spec does allow both types. We should revisit this if it was supposed to be otherwise.

Reply:
(fixed a typo...)

Reply:
Previous discussion at https://github.com/data-apis/array-api/pull/696#issuecomment-1754325857

Reply:
That discussion is for `rfft` and friends. But Leo is saying that the main `fft` should only accept `complex`, not `float`.